packages/rpc-core/src/rpc-methods/common.ts

```diff
@@ -10,6 +10,13 @@ export type DataSlice = Readonly<{
 // the JSON-RPC transport.
 export type LamportsUnsafeBeyond2Pow53Minus1 = bigint & { readonly __lamports: unique symbol };
 
+// FIXME(solana-labs/solana/issues/30341) Beware that any value above 9007199254740991 may be
+// truncated or rounded because of a downcast to JavaScript `number` between your calling code and
+// the JSON-RPC transport.
+//
+// Spefically being used to denote micro-lamports, which are 0.000001 lamports.
+export type MicroLamportsUnsafeBeyond2Pow53Minus1 = bigint & { readonly __lamports: unique symbol };
```

Comment:
Ooh, the brand needs to be different than the one above.

```diff
- export type MicroLamportsUnsafeBeyond2Pow53Minus1 = bigint & { readonly __lamports: unique symbol };
+ export type MicroLamportsUnsafeBeyond2Pow53Minus1 = bigint & { readonly __microLamports: unique symbol };
```

Reply:
Nice catch, put in a PR now so I didn't forget #1527